chore(meta): Updates roles and nomination processes

dipikabh opened the pull request:
This PR rewrites the role descriptions based on internal discussions.

TO DOs:

- [ ] "Processes" sections need to be redone based on actual steps
- [x] **Question**: 'Spotlight contributor' has been moved to 'Specialized roles' because it does not sit well with other 'Roles'. One idea is to change the title 'Specialized roles' to 'Special roles' so that it can include spotlight contributor, invited expert, and community manager better. (Update: done in https://github.com/mdn/content/pull/22916/commits/4cdd04fdd1d70db6ab0ec780f1f89522f9a92b31)
- [ ] **Question**: What info should be retained from [Volunteer and partner maintainers](https://developer.mozilla.org/en-US/docs/MDN/Community/Users_teams#volunteer_and_partner_maintainers)? Should we use the term 'partner maintainer' or is 'maintainer' good enough?
- [x] Add link to pull request guidelines if they are finalized before landing this (https://github.com/mdn/content/pull/22916/commits/8b9efccc051901303bf6a6599c49550d6aeb8c08)
- [ ] Add discord info for contacting the MDN team
- [x] [CONTRIBUTOR_ROLES.md](https://github.com/mdn/project-template/blob/main/CONTRIBUTOR_ROLES.md) to be removed: https://github.com/mdn/project-template/pull/21

AFTER THIS PR LANDS:
- [x] File name to be updated: users_teams --> roles_teams: https://github.com/mdn/content/pull/23534
- [x] Slug to be updated: MDN/Community/Users_teams --> MDN/Community/Roles_teams: https://github.com/mdn/content/pull/23534
- [ ] Sidebar to be updated; https://github.com/mdn/yari/pull/7943

dipikabh committed:
Updates roles and nomination processes

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+
+### Contributor
+
+Contributors, or _community participants_, engage with the project and its community by contributing their time, skills, opinions, and ideas.
```
I've already read through most of this already, the only improvement I can think of is to simplify some of the language so it's not too wordy:

```suggestion
Contributors, or _community participants_, add to the project with their time, skills, opinions, and ideas.
```

dipikabh replied:
Agree!!

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+
+Contributors, or _community participants_, engage with the project and its community by contributing their time, skills, opinions, and ideas.
+Contributors work on the project directly and add value to it.
+The criteria for what counts as a contribution is not limited to writing or testing code, but also includes creating and updating documentation, researching, fixing bugs, and helping other community members.
```
Simplify:

```suggestion
Aside from writing and testing code, contributions include creating and updating documentation, researching, fixing bugs, and helping other community members.
```

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
 
-A list of the members of the organization can be [found here](https://github.com/orgs/mdn/people).
+Community contributions help this open source project immensely.
+In return, contributors can use their work on this project to demonstrate their writing and technical skills as well as their ability to collaborate in team settings and engage with people from diverse backgrounds. This section describes the different roles you can assume while volunteering on the MDN Web Docs project.
```
Simplify maybe?

```suggestion
Contributors can use their work on MDN Web Docs to show writing, collaboration, technical skills, and the ability to work with people from diverse backgrounds. This document describes the roles you can take while volunteering on the MDN Web Docs project.
```

dipikabh replied:
keeping "this section" because this document now contains a roles section and a teams section

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+
+The contributor roadmap is a step-by-step journey from one role to the next as you take on more responsibilities in the project.
+With the advancement in the contributor roadmap, you could serve more than one role at the same time.
+In some cases, such as with the [Invited experts](#invited_expert) role, it's possible to skip the stepwise progression and assume a role directly if you've demonstrated expertise in a particular area.
```
Maybe simplify

```suggestion
Roles such as [Invited experts](#invited_expert) can be directly obtained if you've demonstrated expertise in a particular area.
```

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+Community managers share many of the same responsibilities as a [maintainer](#maintainer).
+In addition, community managers have the following responsibilities:
+
+- Addressing reports of violation of code of conduct and deciding on the appropriate action.
```
A link might be helpful

```suggestion
- Addressing reports of violation of [Mozilla's code of conduct](CODE_OF_CONDUCT.md) and deciding on the appropriate action.
```

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+See who can be a [spotlight contributor](#spotlight_contributor).
+To nominate someone as a spotlight contributor, open an issue on GitHub:
+
+1. Go to [`Issues`](https://github.com/mdn/mdn/issues) in the `mdn/mdn` repository.
```
Same comment: https://github.com/mdn/mdn/issues/new/choose

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+
+To nominate someone as an invited expert, open an issue on GitHub:
+
+1. Go to [`Issues`](https://github.com/mdn/mdn/issues) in the `mdn/mdn` repository.
```
Same 😄  https://github.com/mdn/mdn/issues/new/choose

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+- Once the pull request is approved, the new Invited Expert is added to the [invited-experts team](https://github.com/orgs/mdn/teams/invited-experts) and the appropriate topic team.
+-->
+
+### Stepping down voluntarily
```
We're talking about voluntarily stepping down in the title, but the section is about inactivity. I would rename the section to something along the lines of commitment / what activity is.

For the section content, I thought the following, but depending on the title, you can throw this away:

```
Participants must be active to set an example and show commitment to the project.
Inactivity harms the project as it may lead to unexpected delays, contributor attrition, and a loss of trust in the project.

Inactivity is defined as:

- Not contributing to the project for at least six months.
- Not responding to communication for at least three months.

Inactivity results in a community role being revoked or downgraded and the member is moved to [emeritus status](#gaining-emeritus-status).
```

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,304 @@ tags:
+-->
+
+### Stepping down voluntarily
+
```
I believe we still need a "Stepping down" section where we state that it's okay to step away from roles due to any reason. We want participants to be able to let us know if they are overwhelmed, need a break, need to downgrade to a less-demanding role, or need more help, and which channels are best to talk to us etc.

dipikabh replied:
Valid feedback here 👍.

- I've merged the sections on "Stepping down voluntarily" and "Gaining emeritus status" under "Stepping down or applying for emeritus status"
- I've renamed "Getting demoted involuntarily" to "Demoting or removing inactive contributors" and reorganized the content in this section

Let me know what you think

bsmth replied:
Looks good 👍🏻

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -35,7 +327,7 @@ A list of the members of the organization can be [found here](https://github.com
   - **Yari content SVG**: Responsible for reviewing docs under `/en-US/docs/Web/SVG`
```
Do we need to list all of these? Is it useful for this doc?

dipikabh replied:
On one hand, we could remove the list and point to the mdn/teams link on GitHub. That will reduce the burden of updating this list here should any change happen to the teams (addition, removal).
On the other hand, explicitly listing the teams here might increase the likelihood of interested contributors to spot exactly the team they could help with for reviewing content (just an assumption).

dipikabh replied:
Valid feedback here 👍.

- I've merged the sections on "Stepping down voluntarily" and "Gaining emeritus status" under "Stepping down or applying for emeritus status"
- I've renamed "Getting demoted involuntarily" to "Demoting or removing inactive contributors" and reorganized the content in this section

Let me know what you think

dipikabh replied:
Thanks for the idea! Shortened the list and also updated team names as per https://github.com/orgs/mdn/teams. Fix in https://github.com/mdn/content/pull/22916/commits/565842e3c057d25994bc44cd2e71618c5cf466da.

bsmth replied:
Looks good!

bsmth commented:
Thanks Dipika, when we're nearly ready to merge, we should have a scan over the doc from a readability perspective, my first thoughts were to reduce the complexity of the language a little so it's less intimidating. 👍🏻

dipikabh committed:
Fixes review comments part 1

dipikabh committed:
Fixes review comments part 2

dipikabh committed:
adds roles table

dipikabh commented:
In https://github.com/mdn/content/pull/22916/commits/4cdd04fdd1d70db6ab0ec780f1f89522f9a92b31:
- changed title from "Contributor roles and teams" to "Roles and teams on MDN"
- added a table to compare and summarize the roles (section "Summary of the roles")
- changed the section heading "Specialized roles" to "Special roles"

bsmth committed:
Merge branch 'main' into contrib-roles

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,12 +10,307 @@ tags:
+
+### Summary of the roles
+
+||[Contributor](#contributor)|[Organization member](#organization_member)|[Maintainer](#maintainer)|[Owner](#owner)|
```
What do you think about flipping the orientation to:

| Role                    | Requirements                                                                                                                                                                                                                                                                | Privileges                                                                                               |
| ----------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- |
| **Contributor**         | Follow the code of conduct and contribution guidelines                                                                                                                                                                                                                      | - Invitations to contributor events<br>- Eligibility to become an organization member                    |
| **Organization member** | - Enable 2FA for GitHub account<br>- Enable signed commits<br><br>One or more of:<br>- Resolve two or more issues<br>- Contribute for at least two months<br>- Active contribution in a project area                                                                        | Access rights at organization level                                                                      |
| **Maintainer**          | One or more of:<br>- Invited expert for at least six months<br>- Knowledge across multiple project areas<br>- Act towards overall health of the project<br>- Mentor other contributors<br>- Spend at least 16 hours per month on the project<br>- Attend community meetings | Approve and merge pull requests                                                                          |
| **Owner**               | Limited to Mozilla staff                                                                                                                                                                                                                                                    | - Manage access of different roles to various repositories<br>- Add or archive repositories and projects |

dipikabh replied:
Makes better sense this way, thanks! Fixed in https://github.com/mdn/content/pull/22916/commits/4c82f8a3a02a5c1ed6747140fea98a83db67984d

dipikabh committed:
transposes rows and cols in roles table

dipikabh committed:
shortens team list

dipikabh committed:
Merge branch 'main' into contrib-roles

dipikabh committed:
adds updated link to pull request page

dipikabh commented:
Thanks a lot @bsmth for reviewing this page. Let me know if there's anything else that can be improved.
There are a few pending [TO DOs](https://github.com/mdn/content/pull/22916#issue-1493356007). I can take them up in a subsequent PR.

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,113 +10,334 @@ tags:
 {{MDNSidebar}}
 
-The success and growth of MDN Web Docs is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. These people are part o ...
+The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. To manage a ...
```
We have 2x "There are also" sentences here, maybe we want to fix it in this PR.

dipikabh replied:
Done in https://github.com/mdn/content/pull/22916/commits/d34e7983411723fb927a843269909b2e2900cb76. Thanks for pointing it out.

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,113 +10,334 @@ tags:
 {{MDNSidebar}}
 
-The success and growth of MDN Web Docs is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. These people are part o ...
+The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. To manage a ...
```
Maybe like this?

```suggestion
The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. Changes to the site are performed by employees, contractors, and a network of partners dedicated to the health, growth, and maintenance of the project. Some individuals have committed a portion of their time to assist with running MDN Web Docs. The project relies heavily on [roles](#roles) and [teams](#teams) in the [MDN organization on GitHub](https://github.com/mdn) to manage and incorporate changes from these different groups. A list of the organization's members can be [found here](https://github.com/orgs/mdn/people).
```

dipikabh replied:
Oops, saw this after the previous commit. I've incorporated some of these edits. Thanks for suggestions!

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,113 +10,334 @@ tags:
+- **sre**: Site reliability engineers who support MDN Web Docs
+- **yari-content**: The umbrella team for all MDN Web Docs content reviewers
+
+  - There is a sub-team for the different topic areas — accessibility, Add-ons, CSS, HTML, HTTP, Javascript, SVG, Web API, and Web assembly. For example, there's yari-content-css and yari-content-svg.
```
```suggestion
  - There is a sub-team for the different topic areas — accessibility, Add-ons, CSS, HTML, HTTP, Javascript, SVG, Web API, and Web assembly. For example, there's `yari-content-css` and `yari-content-svg`.
```

We could also do `@yari-content-css` variant as I guess that's how they're tagged on GitHub

dipikabh replied:
That's a good suggestion. I added the '@' to all other team names and removed the bold formatting https://github.com/mdn/content/pull/22916/commits/fd586853a57a99b8d8afa2484cda7799f57f1c6d

bsmth commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -10,113 +10,334 @@ tags:
+- **yari-content**: The umbrella team for all MDN Web Docs content reviewers
+
+  - There is a sub-team for the different topic areas — accessibility, Add-ons, CSS, HTML, HTTP, Javascript, SVG, Web API, and Web assembly. For example, there's yari-content-css and yari-content-svg.
+  - There are also sub-teams for different languages — Brazilian Portuguese, Chinese, French, Japanese, Korean, Russian, and Spanish. For example, there's yari-content-fr and yari-content-ko.
```
Same comment here

bsmth commented:
Thanks @dipikabh, I had another read through and found some quick edits we can do, the rest we can look at after it's merged. Thanks for the overhaul on this 🙌🏻

dipikabh committed:
Merge branch 'main' into contrib-roles

dipikabh commented on files/en-us/mdn/community/users_teams/index.md:
```diff
@@ -1,5 +1,5 @@
 ---
-title: Owners and teams
+title: Roles and teams on MDN
```
Keeping the order as roles first and then teams though to match the order of content.

dipikabh committed:
rephrases para

bsmth reviewed (approved):
Thanks, Dipika! 🙌🏻

dipikabh committed:
updates team mentions

dipikabh committed:
fixes review comments
